EngineTests/AI/Pathing/BinaryMinHeapTest.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using C7Engine.Pathing;
+using Xunit;
+
+namespace EngineTests {
+	public class BinaryMinHeapTest {
+		private void checkBunchInsert<T>(List<T> list) where T:IComparable<T> {
+			BinaryMinHeap<T> heap = new BinaryMinHeap<T>();
+			foreach (T v in list) {
+				heap.insert(v);
+			}
+
+			Assert.Equal(list.Count, heap.count);
+
+			List<T> result = new List<T>();
+			while (heap.count > 0) {
+				result.Add(heap.extract());
+			}
+			Assert.Equal(list.Count, result.Count);
+
+			List<T> sortedCopy = new List<T>(list);
+			sortedCopy.Sort();
+			Assert.Equal(result, sortedCopy);
+		}
+
+		public List<int> generateLargeList() {
+			List<int> lst = new List<int>();
+			for (int i = 0; i < 1000; ++i) {
+				lst.Add(i);
+			}
+
+			Random random = new Random(0x1337);
```

Comment:
I see how that random seed was chosen...

One minor issue, the first 39 lines of this file have spaces for indentation, whereas the rest of the PR uses tabs.  We've still got some legacy spaces in other files, but are trying to standardize on tabs.

Reply:
Are you sure? I checked this file now and found nothing but tabs.

Reply:
Odd, it must have been a GitHub glitch, when I view it in Rider it is indeed tabs. Sorry for the false alarm!